src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -223,6 +224,51 @@ protected override ObjectTypeConstructor BuildSerializationConstructor()
                 serializationInitializers.Add(new ObjectPropertyInitializer(RawDataField, deserializationParameter, null));
             }
 
+            if (ModelTypeMapping != null)
+            {
+                foreach (var propertyWithSerialization in ModelTypeMapping.GetPropertiesWithSerialization())
+                {
+                    if (serializationInitializers.Exists(i => i.Property.Declaration.Name == propertyWithSerialization.Name))
+                        continue;
+                    var csharpType = BuilderHelpers.GetTypeFromExisting(propertyWithSerialization, typeof(object), MgmtContext.TypeFactory);
+                    var isReadOnly = propertyWithSerialization switch
+                    {
+                        IPropertySymbol propertySymbol => propertySymbol.SetMethod == null,
+                        IFieldSymbol fieldSymbol => fieldSymbol.IsReadOnly,
+                        _ => throw new NotSupportedException($"'{propertyWithSerialization.ContainingType.Name}.{propertyWithSerialization.Name}' must be either field or property.")
+                    };
+                    var xml = propertyWithSerialization.GetDocumentationCommentXml();
+                    string comment = "";
+                    if (!string.IsNullOrEmpty(xml))
+                    {
+                        var xd = new global::System.Xml.XmlDocument();
+                        xd.LoadXml(xml);
+                        comment = xd.SelectSingleNode("/member/summary")?.InnerText.Trim() ?? "";
+                    }
+                    var otp = new ObjectTypeProperty(
+                        new MemberDeclarationOptions(propertyWithSerialization.DeclaredAccessibility.ToString(), propertyWithSerialization.Name, csharpType),
+                        comment,
+                        isReadOnly,
+                        new Property()
+                        {
+                            Required = true,
+                            ReadOnly = isReadOnly,
+                            IsDiscriminator = false,
+                            SerializedName = propertyWithSerialization.Name,
+                        });
```

Comment:
This is the place I believe we should assign a property `Schema` for this made-up `Property` because the serialization builder needs this to identify if this should be serialized as a list or dictionary or normal property.
Just make a new method following the same pattern as we have in `ModelTypeProviderFields.GetInputTypeFromExistingMemberType` method to construct a new schema for it.

Reply:
Do you mean have a switch for list/dict/object/string/int/binary/... to generate corresponding schema? any existing code to do so? or any sample code i can refer to cover all the possibility of the switch?

Reply:
I need to correct my previous statement: we actually do not need a `Property` here. The `SchemaProperty` property in `ObjectTypeProperty` is nullable, we just need to put it as null here.
This is the same thing we did for `ModelTypeProvider`.
You need to go over how we did it for `ModelTypeProvider` and use the same approach for `SchemaObjectType`.

How the property is built: https://github.com/Azure/autorest.csharp/blob/23dcd81c26074c182961ae4a10319915ec4026ff/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs#L132-L151
How the serialization is constructed: https://github.com/Azure/autorest.csharp/blob/23dcd81c26074c182961ae4a10319915ec4026ff/src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs#L302-L317
In `ModelTypeProvider` we just put the `InputModelProperty` property as null value, to indicate that this property is not included in the spec (which is the case you are trying to solve: add a new property to serialization, otherwise it is "changing the serialization of an existing property")

Reply:
update to null. and found additionalProperties would also have schemaProperty=null. Some check is also added for that.